src/Data/Monoid/Action/RightAction.hs

```diff
@@ -0,0 +1,69 @@
+{-# LANGUAGE FlexibleInstances     #-}
+{-# LANGUAGE MultiParamTypeClasses #-}
+
+-----------------------------------------------------------------------------
+-- |
+-- Module      :  Data.Monoid.Action.RightAction
+-- Copyright   :  (c) 2011 diagrams-core team (see LICENSE)
+-- License     :  BSD-style (see LICENSE)
+-- Maintainer  :  diagrams-discuss@googlegroups.com
+--
+-- Monoid and semigroup right actions.
+--
+-----------------------------------------------------------------------------
+
+module Data.Monoid.Action.RightAction
+       ( RightAction(..)
+       ) where
+
+import           Data.Semigroup
+
+------------------------------------------------------------
+--  Monoid and semigroup right actions
+------------------------------------------------------------
+
+-- | Type class for right monoid (and semigroup) actions, where monoidal
+--   values of type @m@ act (@rightAct@) on values of another type @s@.
+--   Instances are required to satisfy the laws
+--
+--   * @mempty \`rightAct\` s = s@
```

Comment:
These laws need to be fixed, I think they are copy-pasted from the left action laws.